data/mods/StatsThroughSkills/main.lua

```diff
@@ -0,0 +1,393 @@
+--Increasing stats by skill
+local MOD = {}
+
+mods["StatsThoughSkills"] = MOD
+
+function MOD.on_day_passed()
+    game.add_msg("Calculating new stats based off skills")
+    int_bonus = 8
+    str_bonus = 8
+    dex_bonus = 8
+    per_bonus = 8
+
+    --based skills
+    barter_skill = player:get_skill_level("barter")
+    if (barter_skill >= 3) then
+        int_bonus = int_bonus + 1
+        if (barter_skill >= 6) then
+            int_bonus = int_bonus + 1
+            if (barter_skill >= 9) then
+                int_bonus = int_bonus + 1
+            end
+        end
+    end
+
+    computer_skill = player:get_skill_level("computer")
+    if (computer_skill >= 3) then
+        int_bonus = int_bonus + 1
+        if (computer_skill >= 6) then
+            int_bonus = int_bonus + 1
+            if (computer_skill >= 9) then
+                int_bonus = int_bonus + 1
+            end
+        end
+    end
+
+    cooking_skill = player:get_skill_level("cooking")
```

Comment:
Am I the only one who things that all those blocks look identical and should go into a separate function?

```Lua
int_bonus = calc_skill_bonus(int_bonus, "cooking")
```
Reads far nicer and is easier to understand.